rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -470,15 +496,153 @@ private void mergeRequestedDependencies(List<Dependency> incomingRequestedDepend
             }
         }
 
+        @Value
+        private class PluginKey {
+            String groupId;
+            String artifactId;
+        }
+
+        private PluginKey getPluginKey(Plugin plugin) {
+            return new PluginKey(
+                    plugin.getGroupId(),
+                    plugin.getArtifactId()
+            );
+        }
+
+        private List<Dependency> mergePluginDependencies(List<Dependency> dependencies, List<Dependency> incomingDependencies) {
+            if (incomingDependencies.isEmpty()) {
+                return dependencies;
+            }
+            if (dependencies.isEmpty()) {
+                return incomingDependencies;
+            }
+
+            List<Dependency> merged = new ArrayList<>();
+            Set<GroupArtifact> uniqueDependencies = new HashSet<>();
+            for (Dependency dependency : dependencies) {
+                merged.add(dependency);
+                uniqueDependencies.add(new GroupArtifact(dependency.getGroupId(), dependency.getArtifactId()));
+            }
+            for (Dependency dependency : incomingDependencies) {
+                if (!uniqueDependencies.contains(new GroupArtifact(dependency.getGroupId(), dependency.getArtifactId()))) {
+                    merged.add(dependency);
+                }
+            }
```

Comment:
deduplicating at merging time, as opposed to dependencies, since I think it's better than adding the logic to `deduplicate()`